src/VisualStudio/Core/Def/Implementation/Log/VisualStudioErrorLogger.cs

```diff
@@ -51,36 +53,7 @@ public bool TryLogException(object source, Exception exception)
             return watsonReportResult && activityLogResult;
         }
 
-        private bool ShouldReportCrashDumps(object source)
-        {
-            var refactoringAttributes = source.GetType()
-                .GetCustomAttributes(typeof(ExportCodeRefactoringProviderAttribute), true)
-                .Cast<ExportCodeRefactoringProviderAttribute>();
-            if (PredefinedCodeRefactoringProviderNames.Any(name => refactoringAttributes.Select(x => x.Name).Any(n => n == name)))
-            {
-                return true;
-            }
-            var codeFixAttributes = source.GetType()
-                .GetCustomAttributes(typeof(ExportCodeFixProviderAttribute), true)
-                .Cast<ExportCodeFixProviderAttribute>();
-            if (PredefinedCodeFixProviderNames.Any(name => refactoringAttributes.Select(x => x.Name).Any(n => n == name)))
-            {
-                return true;
-            }
-
-            return false;
-        }
-
-        private static IEnumerable<string> PredefinedCodeRefactoringProviderNames => s_lazyPredefinedCodeRefactoringProviderNames.Value;
-
-        private static Lazy<IEnumerable<string>> s_lazyPredefinedCodeRefactoringProviderNames =
-            new Lazy<IEnumerable<string>>(() => typeof(PredefinedCodeRefactoringProviderNames).GetFields().Select(f => f.GetRawConstantValue().ToString()));
-
-
-        private static IEnumerable<string> PredefinedCodeFixProviderNames => s_lazyPredefinedCodeFixProviderNames.Value;
-
-        private static Lazy<IEnumerable<string>> s_lazyPredefinedCodeFixProviderNames =
-            new Lazy<IEnumerable<string>>(() => typeof(PredefinedCodeFixProviderNames).GetFields().Select(f => f.GetRawConstantValue().ToString()));
+        private bool ShouldReportCrashDumps(object source) => HasRoslynAssemblyAttribute(source);
```

Comment:
Was the old function only taking the object to get the type? We should make this method more strongly typed then.